sdk_extensions/logging/support/src/main/java/io/opentelemetry/common/AnyValue.java

```diff
@@ -0,0 +1,276 @@
+/*
+ * Copyright 2020, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.common;
+
+import com.google.auto.value.AutoValue;
+import java.util.List;
+import java.util.Map;
+import javax.annotation.Nullable;
+import javax.annotation.concurrent.Immutable;
+
+/**
+ * A class that represents all the possible values for a data body. An {@code AnyValue} can have 6
+ * types of values: {@code String}, {@code boolean}, {@code int}, {@code double}, {@code array}, or
+ * {@code kvlist}. represented through {@code AnyValue.Type}. A {@code array} or a {@code kvlist}
+ * can in turn hold other {@code AnyValue} instances, allowing for mapping to JSON-like structures.
+ *
+ * @since 0.8.0
+ */
+@Immutable
+public abstract class AnyValue {
```

Comment:
This may need to be replaced by `AttributeValue`, I think the missing part is the `KVList`, maybe a comment for the moment explaining the difference.

Reply:
This is a good point. In Collector codebase for example we use the same AttributeValue data type for Spans, Resources and LogRecords. This mirrors what we have in the OTLP Proto definitions.

Is there any reason we cannot extend [AttributeValue](https://github.com/open-telemetry/opentelemetry-java/blob/master/api/src/main/java/io/opentelemetry/common/AttributeValue.java) to fully handle all possible types that AnyValue can represent?

Reply:
If the recent PR #1631 gets merged, then AttributeValue won't exist any more. Also, I don't think we want to accept map-values into our tracing attributes, which that would impact.